pkg/cmd/roachtest/tests/mixed_version_cdc.go

```diff
@@ -561,3 +586,67 @@ func runCDCMixedVersions(ctx context.Context, t test.Test, c cluster.Cluster) {
 	mvt.AfterUpgradeFinalized("wait and validate", tester.waitAndValidate)
 	mvt.Run()
 }
+
+// runCDCMixedVersionCheckpointing tests that the new span-level checkpoint
+// code added in 25.2 works correctly in mixed-version states.
+//
+// Writing the checkpoint is explicitly forced with cluster settings and
+// restoring the checkpoint implicitly happens following each of the rolling
+// restarts during the mixed-version test run.
+func runCDCMixedVersionCheckpointing(ctx context.Context, t test.Test, c cluster.Cluster) {
+	tester := newCDCMixedVersionTester(ctx, c)
+
+	mvt := mixedversion.NewTest(
+		ctx, t, t.L(), c, tester.crdbNodes,
+		// We're only concerned with mixed-version compatibility starting at
+		// versions that can upgrade to 25.2 (only 24.3 and 25.1), since that's
+		// the first version with the new span-level checkpoint format.
+		mixedversion.MinimumBootstrapVersion("v24.3.0"),
+	)
+
+	cleanupKafka := tester.StartKafka(t, c)
+	defer cleanupKafka()
+
+	forceCheckpointing := func(
+		ctx context.Context, l *logger.Logger, r *rand.Rand, h *mixedversion.Helper,
+	) error {
+		// NB: We use the 24.3 names for the cluster settings so that the cluster
+		// can understand them at startup time.
+		for _, stmt := range []string{
+			`SET CLUSTER SETTING changefeed.frontier_checkpoint_frequency = '1s'`,
+			`SET CLUSTER SETTING changefeed.frontier_highwater_lag_checkpoint_threshold = '1us'`,
+		} {
+			if err := h.System.Exec(r, stmt); err != nil {
```

Comment:
`changefeed.frontier_checkpoint_frequency` is an application level cluster setting. Does it make sense to set it on the system tenant in multitenant deployments? More of an educational question (if intentional).